### PR TITLE
[docs] correct code highlight and code comments in Learn section

### PR DESCRIPTION
**Pull request by Simek**
# Why

Several example looks off and even broken after upgrading Prism to the latest version.

# How

Update code example and code comments in the Snippets used on the Learn section pages.

Now, our custom comments need to be a valid in syntax comments in some cases, and cannot be used one directly after another because Prism parses them as a single plain text block. Also we cannot comment on JSX component props directly, since for this we need to use `{/* ... */}` notation too, which will leave the curly braces in the output.

Not all fixes are ideal, when it come to highlighted code portion, but it's the best what I was able to do without breaking the code syntax highlight at all.

# Test Plan

The changes have been reviewed locally. All lint checks and tests are passing.

# Preview

Screenshot 2024-01-10 at 17 56 26

### CODE REVIEW

**Review by amandeepmittal — APPROVED** [review]
Thanks for fixing this!

File: docs/pages/tutorial/build-a-screen.mdx [hunk 1]
```diff
@@ -377,7 +380,7 @@ const styles = StyleSheet.create({
 });
 ```
 
-Let's take a look at our app on iOS, Android and the web:
+Let's take a look at our app on Android, iOS and the web:
```
Comment (amandeepmittal) [inline docs/pages/tutorial/build-a-screen.mdx:383]: One small suggestion to revert this change is because if you look at the actual screenshot (or video in some other but similar cases) it refers to the actual platforms shown in that order.

TLDR: it was intentional.

File: docs/pages/tutorial/introduction.mdx [hunk 2]
```diff
@@ -73,7 +73,7 @@ const styles = StyleSheet.create({
 >
 > <br />
 > Go ahead and press the above button. You will see the above code running in a Snack. Try to switch
-> between iOS, Android, or the web tabs. You can also open it on your device in the Expo Go app from
+> between Android, iOS or the web tabs. You can also open it on your device in the Expo Go app from
```
Comment (amandeepmittal) [inline docs/pages/tutorial/introduction.mdx:76]: Snack also uses iOS, Android, Web order so let's stick to it here in content of this info supporting the actual UI of the platform.

Comment (Simek) [reply docs/pages/tutorial/introduction.mdx:76]: Let's change the order in Snack then too, so it matches our guidelines. 😉
